라이브러리 추가 및 JDBC 연동

sominseob opened the pull request:
의존성 주입 및 jdbc 연결

sominseob committed:
의존성 주입 및 jdbc 연결

KimKiHyun0206 commented on src/test/java/test.java:
```diff
@@ -0,0 +1,34 @@
+                con.close();
+        } catch (SQLException e) {}
+    }
+}
```
test에 쓰지 말고 프로젝트에 구현해주세요

Duskafka reviewed (commented):
테스트 코드는 코드를 구현하는 곳이 아이라 구현된 코드를 테스트하는 곳입니다.

Duskafka commented on src/test/java/test.java:
1. 테스트 코드에서는 main 메소드를 사용하면 안됩니다.
2. test 에서는 구현된 코드를 테스트하는 곳입니다

sominseob replied:
@test로 해서 작성하겠습니다!

Duskafka commented on src/test/java/test.java:
```diff
@@ -0,0 +1,34 @@
+
+        // 접속
+        try {
+            con = DriverManager.getConnection(url, user_name, password);
```
user_name이라는 변수명은 카멜케이스로 수정이 필요합니다

sominseob replied:
userName 으로 변경했습니다

Duskafka commented on src/test/java/test.java:
```diff
@@ -0,0 +1,34 @@
+        Connection con = null;
+
+        String url = "jdbc:mysql://localhost:3306/quiz_db"; // 서버 주소
+        String user_name = "root"; //  접속자 id
```
user_name이라는 변수명은 카멜케이스로 수정이 필요합니다

sominseob replied:
userName 으로 변경했습니다

Duskafka commented on src/test/java/test.java:
```diff
@@ -0,0 +1,34 @@
+import java.sql.*;
+
+public class test {
+    public static void main(String[] args) {
```
테스트 코드 작성은 main으로 하지 않고 @Test 애노테이션으로 합니다

sominseob replied:
main 메소드 사용안하고 @test로 바꿨습니다!

Duskafka commented on src/test/java/test.java:
```diff
@@ -0,0 +1,34 @@
+        try {
+            if(con != null)
+                con.close();
+        } catch (SQLException e) {}
```
캐치 후의 처리가 없는데 구현이 필요합니다

sominseob replied:
캐치문 내부 작성완료했습니다!

Duskafka commented on src/test/java/test.java:
```diff
@@ -0,0 +1,34 @@
+        try {
+            Class.forName("com.mysql.cj.jdbc.Driver");
+        } catch (ClassNotFoundException e) {
+            System.err.println("JDBC 드라이버를 로드하는데에 문제 발생" + e.getMessage());
```
println보다는 로그를 사용해주세요

sominseob replied:
loger 사용해서 다시 작성했습니다.

Duskafka commented on src/main/resources/application.properties:
```diff
@@ -1 +1,6 @@
+spring.datasource.url=jdbc:mysql://localhost:3306/quiz_db
+spring.datasource.username=root
+spring.datasource.password=0000
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
```
프로그램 내에서 이 값들을 사용하지 않는 것 같은데 유지할 필요가 있을까요

sominseob replied:
삭제했습니다.

sominseob committed:
jdbc 연결

-application.properties 쓸모없는 코드 삭제

-main -> @Test 애노테이션 변경

-user_name -> userName 변경

println -> logger 사용

catch 이후 코드 작성

Duskafka commented on src/test/java/test.java:
```diff
@@ -1,34 +1,53 @@
 
 public class test {
-    public static void main(String[] args) {
+    private static final Logger logger = LoggerFactory.getLogger(test.class);
```
@Slf4j 애노테이션을 클래스 위에 사용하면 클래스를 생성하지 않아도 log 애노테이션을 사용할 수 있습니다

sominseob replied:
로그 사용해서 다시 했습니다

Duskafka commented on src/test/java/test.java:
```diff
@@ -1,34 +1,53 @@
-            System.err.println("연결 오류" + e.getMessage());
-            e.printStackTrace();
+        while (retryCount < MAX_RETRY_COUNT) {
+            try {
+                con = DriverManager.getConnection(url, userName, password);
+                logger.info("연결 완료!");
+                break;
```
왜 break가 들어갔는지 설명해주세요

sominseob replied:
변경했습니다.

Duskafka commented on src/test/java/test.java:
```diff
@@ -1,34 +1,53 @@
+                logger.error("연결 오류 발생. 재시도 중...", e);
+                if (retryCount == MAX_RETRY_COUNT) {
+                    logger.error("최대 재시도 횟수 초과. 연결에 실패했습니다.");
+                    return;
```
왜 return이 들어갔는지 알려주세요

sominseob replied:
변경 했습니다

sominseob committed:
jdbc 연결

test Test로 바꿀려했는데 충돌나서 나중에 JdbcConnectionTest 로 바꾸거나 하겠습니다.

while문 -> try-catch로 바꿨습니다.
나머지 변경 사항 번경했습니다.

Duskafka reviewed (commented):
Assertion에 대해서 공부하고 테스트 코드에 적용해보세요
